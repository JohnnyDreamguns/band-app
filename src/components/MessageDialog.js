import React from 'react';

const MessageDialog = ({ messages }) => (
  <React.Fragment>
    {messages.error && (
      <div className={'notify-box error'}>{messages.error}</div>
    )}
    {messages.info && <div className={'notify-box info'}>{messages.info}</div>}
  </React.Fragment>
);

export default MessageDialog;
