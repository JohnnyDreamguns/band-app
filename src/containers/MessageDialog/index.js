import React from 'react';
import { connect } from 'react-redux';
import { selectMessages } from './selectors';

const MessageDialog = ({ messages }) => (
  <React.Fragment>
    {messages.error && (
      <div className={'notify-box error'}>{messages.error}</div>
    )}
    {messages.info && <div className={'notify-box info'}>{messages.info}</div>}
  </React.Fragment>
);

export const mapStateToProps = state => {
  return {
    messages: selectMessages(state)
  };
};

export default connect(mapStateToProps)(MessageDialog);
