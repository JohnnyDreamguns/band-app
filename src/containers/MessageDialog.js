import React from 'react';
import MessageDialog from '../components/MessageDialog';
import { useMessages } from '../hooks/useMessages';

const MessageDialogContainer = () => {
  const { messages } = useMessages();

  const props = {
    messages
  };

  return <MessageDialog {...props} />;
};

export default MessageDialogContainer;
