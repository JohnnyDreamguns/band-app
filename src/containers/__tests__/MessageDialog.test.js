import React from 'react';
import TestRenderer from 'react-test-renderer';
import MessageDialogContainer from '../MessageDialog';
import MessageDialog from '../../components/MessageDialog';

jest.mock('../../hooks/useMessages', () => ({
  useMessages: () => ({
    messages: 'mock messages',
    setError: 'mock setError',
    setErrorObject: 'mock setErrorObject',
    setInfo: 'mock setInfo'
  })
}));

describe('HomePage container', () => {
  const testRenderer = TestRenderer.create(<MessageDialogContainer />);
  const testInstance = testRenderer.root;

  it('should match the props sent to the component', () => {
    expect(testInstance.findByType(MessageDialog.type).props).toMatchSnapshot();
  });
});
