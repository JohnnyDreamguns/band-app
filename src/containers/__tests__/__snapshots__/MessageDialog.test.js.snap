// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage container should match the props sent to the component 1`] = `
Object {
  "messages": "mock messages",
}
`;
